### PR TITLE
Inform spellcheck about extension names

### PR DESCRIPTION
**Pull request by ericpromislow**
No associated issue, but I wrote this while looking at the spellcheck notifications in PR #5370 and figured there must be a simple way to add more extensions without having to add more names to `.github/actions/spelling/allow.txt` - this is it.

Also, we should keep `allow.txt` sorted for ease of use.

### CODE REVIEW

**Comment by ericpromislow** [comment]
Note that nothing is currently using `spellcheck-ignore-line`, but PR #5370 will benefit from it, because there will be a name without an attached numeral in `pkg/rancher-desktop/components/NavIconExtension.vue`, and this ignore-directive will work well there. An alternative is to ignore any path-like string, not just ones starting with an IP protocol, but that might be too general.

File: .github/actions/spelling/patterns.txt [hunk 1]
```diff
@@ -141,3 +141,15 @@ regexp?\.MustCompile\(`[^`]*`\)
...
+# Directives to avoid extension-related publisher names and repositories:
+
+# Directives to skip a full line
+^.*spellcheck-ignore-line.*$
+
+# Ignore extension names (usually based on a github repository)
+^\s*\bslug:\s+'.*?/.*',\s*$
+^\s*\bpublisher:\s+\{\s*name:\s*'.?*'.*?\}
+
+# Ignore extension repository account names (no slash in the quoted part)
```
Comment (mook-as) [inline .github/actions/spelling/patterns.txt:154]: Do we want to enforce the lack of a slash (`[^/]*` instead of `.*`)?

**Review by mook-as — CHANGES_REQUESTED** [review]
9fa7b9ebdbf431e25e398f3ffab572b9ee4b6a7e and 0d3f6d00341b4a0ae16b34aaf7b55ff85d7b5a8a should be merged into bed1f39503de77c3de62a60d354a113391ec48af.

File: .github/actions/spelling/patterns.txt (hunk 1 above)
Comment (ericpromislow) [reply .github/actions/spelling/patterns.txt]: I redid the comments to clarify what these lines do.

**Comment by ericpromislow** [comment]
And I squashed all the commits into one.

**Comment by ericpromislow** [comment]
Finally got the new spellcheck to work, and the good news is I need to use the `spellcheck-ignore-line` pseudo-comment

**Review by mook-as — CHANGES_REQUESTED** [review]
Some stray commits (moproxy?) got mixed into this PR.

**Comment by ericpromislow** [comment]
I recreated the branch to get rid of the extraneous commits -- not sure how they got in, probably based the branch on the wrong branch.

File: pkg/rancher-desktop/utils/_demo_marketplace_items.js [hunk 2]
```diff
@@ -172,14 +172,14 @@ export const demoMarketplace = {
     },
     {
       id:                    '',
-      name:                  'Blockly Automation',
+      name:                  'Blockly Automation', // spellcheck-ignore-line
```
Comment (mook-as) [inline pkg/rancher-desktop/utils/_demo_marketplace_items.js:175]: We should probably ignore the whole file, but this is fine for now.

**Review by mook-as — APPROVED** [review]

File: pkg/rancher-desktop/utils/_demo_marketplace_items.js (hunk 2 above)
Comment (ericpromislow) [reply pkg/rancher-desktop/utils/_demo_marketplace_items.js:175]: We can ignore the whole file with a single entry in `.github/actions/spelling/excludes.txt`

**Pull request merged**